pkg/container/container.go

```diff
@@ -141,6 +142,28 @@ func (c *container) SpecAddAnnotations(sb *sandbox.Sandbox, containerVolumes []o
 	// Preserve the sandbox annotations. OCI hooks may re-use the sandbox
 	// annotation values to apply them to the container later on.
 	for k, v := range sb.Annotations() {
+		// The OCI seccomp BPF hook
+		// (https://github.com/containers/oci-seccomp-bpf-hook)
+		// uses the annotation io.containers.trace-syscall as indicator to
+		// attach a BFP module to the process. The recorded syscalls will be
+		// then stored in the output path file (annotation value prefixed with
+		// 'of:'). We now add a custom logic to be able to distinguish
+		// containers within pods in Kubernetes. If we suffix the container
+		// name within the annotation key like this:
+		// io.containers.trace-syscall/container
+		// Then we will rewrite the key to 'io.containers.trace-syscall' if the
+		// metadata name is equal to 'container'. This allows us to trace
+		// containers into distinguishable files.
+		const traceHookPrefix = "io.containers.trace-syscall"
```

Comment:
maybe this is a good candidate for AllowedAnnotations?

Reply:
Yes, the annotation `io.containers.trace-syscall` now has to be part of the allowed annotations, otherwise they will get filtered-out at the sandbox level. PTAL :)